app/src/controllers/user/sagas.js

```diff
@@ -134,71 +139,83 @@ function* fetchUserWorker() {
-    (key) => assignedOrganizations[key].organizationId === organizationId,
-  );
-  const { organizationSlug: defaultOrganizationSlug } = defaultOrganization
-    ? assignedOrganizations[defaultOrganization]
-    : Object.keys(assignedOrganizations)[0];
-
-  let projectKey;
-  let activeOrganization;
 
-  try {
-    const activeOrganizationResponse = yield call(
-      fetch,
-      URLS.organizationList({ slug: targetOrganizationSlug }),
+  const format = getLogTimeFormatFromStorage(userId);
+  yield put(setLogTimeFormatAction(format));
+
+  if (Object.keys(assignedOrganizations).length === 0) {
+    yield put(setActiveProjectKeyAction(null));
+    yield put(
+      redirect({
+        type: ORGANIZATIONS_PAGE,
+      }),
+    );
+  } else {
+    const userSettings = getStorageItem(`${userId}_settings`) || {};
+    const targetActiveProject = urlOrganizationAndProject || userSettings?.activeProject;
+    const { organizationSlug: targetOrganizationSlug, projectSlug: targetProjectSlug } =
+      targetActiveProject || {};
+
+    const { assignmentNotRequired, isAssignedToTargetProject } = yield select(
+      userAssignedSelector(targetProjectSlug, targetOrganizationSlug),
+    );
+
+    const defaultProject = Object.keys(assignedProjects)[0];
+    const {
+      projectSlug: defaultProjectSlug,
+      projectKey: defaultProjectKey,
+      organizationId,
+    } = assignedProjects[defaultProject];
+    const defaultOrganization = Object.keys(assignedOrganizations).find(
+      (key) => assignedOrganizations[key].organizationId === organizationId,
     );
+    const { organizationSlug: defaultOrganizationSlug } = defaultOrganization
+      ? assignedOrganizations[defaultOrganization]
+      : Object.keys(assignedOrganizations)[0];
 
-    activeOrganization = activeOrganizationResponse?.items?.[0];
-  } catch (e) {} // eslint-disable-line no-empty
+    let projectKey;
+    let activeOrganization;
 
-  if (!isAssignedToTargetProject && assignmentNotRequired) {
     try {
-      // TODO: Fetch project by slug
-      const organizationProjects = yield call(
+      const activeOrganizationResponse = yield call(
         fetch,
-        URLS.organizationProjects(activeOrganization?.id),
+        URLS.organizationList({ slug: targetOrganizationSlug }),
       );
-      projectKey = organizationProjects?.items?.find(({ slug }) => slug === targetProjectSlug)?.key;
+
+      activeOrganization = activeOrganizationResponse?.items?.[0];
     } catch (e) {} // eslint-disable-line no-empty
-  }
 
-  const activeProject =
-    isAssignedToTargetProject || projectKey
-      ? targetActiveProject
-      : { organizationSlug: defaultOrganizationSlug, projectSlug: defaultProjectSlug };
+    if (!isAssignedToTargetProject && assignmentNotRequired) {
+      try {
+        // TODO: Fetch project by slug
+        const organizationProjects = yield call(
+          fetch,
+          URLS.organizationProjects(activeOrganization?.id),
+        );
```

Comment:
Can we add `slug` parameter here to find the project directly by slug?
Looks like API already supports this option
![image](https://github.com/user-attachments/assets/ab714144-a7c0-4a07-b169-d84fd0080170)

Reply:
Done.